src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,171 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int HALF_DIVIDER_INDEX = 2;
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
```

Comment:
Private methods and classes should be after public ones in your class. Move the `Node<T>` class definition to the end of the `MyLinkedList` class.